stubs/sublime_plugin.pyi

```diff
@@ -1,27 +1,35 @@
 import sublime
-from typing import Optional, Union, List, Dict, Tuple
 
+from typing import Callable, Generic, TypeVar, Optional, Union, List, Tuple, overload
 
-class CommandInputHandler():
+
+InputType = TypeVar('InputType')
+
+
+class CommandInputHandler(Generic[InputType]):
     def name(self) -> str: ...
     def next_input(self, args: dict) -> Optional[CommandInputHandler]: ...
     def placeholder(self) -> str: ...
     def initial_text(self) -> str: ...
-    def preview(self, arg: dict) -> Union[str, sublime.Html]: ...
-    def validate(self, arg: dict) -> bool: ...
+    def preview(self, arg: InputType) -> Union[str, sublime.Html]: ...
+    def validate(self, arg: InputType) -> bool: ...
     def cancel(self) -> None: ...
-    def confirm(self, arg: dict) -> None: ...
+
+    @overload
+    def confirm(self, arg: InputType) -> None: ...
+    @overload
+    def confirm(self, arg: InputType, event: dict) -> None: ...
 
 
-class BackInputHandler(CommandInputHandler):
+class BackInputHandler(CommandInputHandler[None]):
     pass
 
 
-class TextInputHandler(CommandInputHandler):
+class TextInputHandler(CommandInputHandler[str]):
     def description(self, text: str) -> str: ...
 
 
-class ListInputHandler(CommandInputHandler):
+class ListInputHandler(CommandInputHandler[InputType], Generic[InputType]):
     def list_items(self) -> list: ...
```

Comment:
Not required for us, most likely, but the return type of `list_items` is much more complex and needs to consider `InputType`. I don't know how detailed we want the stubs to be, though.

cc @jfcherng, since you didn't touch that in https://github.com/jfcherng-sublime/ST-API-stubs/commit/71950113a9473514590b4e720ea3e5886b467d3a#diff-d4b7ec302230de6d2cd37a74fe45305e8eb37f208cc9ab88b2f52301b99b3bd8R716

Reply:
Fixed.